lib/inventory/metadata/metadata_linux.go

```diff
@@ -0,0 +1,71 @@
+//go:build linux
+// +build linux
+
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"fmt"
+	"strings"
+)
+
+// fetchOSVersion combines the content of '/etc/os-release' to be e.g.
+// "Ubuntu 22.04".
+func (c *fetchConfig) fetchOSVersion() string {
+	filename := "/etc/os-release"
+	out, err := c.read(filename)
+	if err != nil {
+		return ""
+	}
+
+	var name string
+	var versionID string
+	for _, line := range strings.Split(strings.TrimSpace(out), "\n") {
+		parts := strings.Split(line, "=")
+		if len(parts) != 2 {
+			return ""
+		}
+
+		switch parts[0] {
+		case "NAME":
+			name = strings.Trim(parts[1], `"`)
+		case "VERSION_ID":
+			versionID = strings.Trim(parts[1], `"`)
+		}
+	}
+
+	return fmt.Sprintf("%s %s", name, versionID)
+}
+
+// fetchGlibcVersion parses the output of 'ldd --version' and returns e.g.
+// "2.31".
+func (c *fetchConfig) fetchGlibcVersion() string {
+	out, err := c.exec("ldd", "--version")
+	if err != nil {
+		return ""
+	}
+
+	lines := strings.Split(strings.TrimSpace(out), "\n")
+	if len(lines) == 0 {
+		return ""
+	}
+
+	// The glibc version is the last word on the first line.
+	parts := strings.Fields(lines[0])
+	return parts[len(parts)-1]
```

Comment:
This will fail quite badly when running in alpine with gcompat. Can we call `gnu_get_libc_version` via cgo instead? I'm pretty sure that teleport doesn't actually compile with non-glibc anyway. cc @zmb3

Reply:
Also added this idea as a possible improvement in https://github.com/gravitational/cloud/issues/3550#issue-1591930330.

Reply:
Launching `ldd` will also fail in the new distroless images, they don't have `ldd`.

Reply:
That also applies to `arch`, btw.

Reply:
Added this to https://github.com/gravitational/cloud/issues/3550#issue-1591930330 as well.

Reply:
After discussing with @espadolini, we're now:
- using `gnu_get_libc_version` (ddcc8c5e2d23e7bd4c911bf0c6c2c1d6cd2d0722) as this seems to be a much more reliable approach
- using `GOARCH` (b694ca9a196b43af1d893df048d68e57fd9b88dc) as there doesn't seem to be a likely situation in which its output would differ from `arch`